Gems/Atom/Tools/MaterialCanvas/Code/Source/Document/MaterialGraphCompiler.cpp

```diff
@@ -235,6 +235,12 @@ namespace MaterialCanvas
 
     void MaterialGraphCompiler::BuildDependencyTables()
     {
+        if (!m_graph)
+        {
+            AZ_Assert(false, "Attempting to generate data from invalid graph object.");
+            return;
```

Comment:
Should this be an AZ_Error - does this error occur from a programming failure or a data failure?